src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  onDeleteTodo: (id: number) => void,
+  loadingTodos: number[],
+  onUpdateTodo: (id: number, data: Partial<Todo>) => void,
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  onDeleteTodo,
+  loadingTodos,
+  onUpdateTodo,
+}) => {
+  const {
+    id,
+    completed,
+    title,
+  } = todo;
+
+  const [inEditMode, setInEditMode] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const isLoading = loadingTodos.includes(id);
+
+  const handleTitleChange = () => {
+    if (!newTitle.trim()) {
+      onDeleteTodo(id);
+    }
```

Comment:
you have the case when is deletes and edits the todo item at the same. you have to add return to not proceed. 

https://user-images.githubusercontent.com/22824431/232745379-2c47729d-6ebe-494f-b06e-d51cd08f4eb5.mov


```suggestion
    if (!newTitle.trim()) {
      onDeleteTodo(id);
      setInEditMode(false);
      return;
    }
```